### DISCUSSION TITLE
MacOS, Bug: Window resizing and jumping behavior when switching/creating tabs in tiling window managers

### DISCUSSION THREAD

**Discussion by mlentzler** [ghostty-org/ghostty#4037] [Feature Requests, Ideas] [answered]
## Description:
When using a tiling window manager like AeroSpace or Amethyst, creating a new tab causes the terminal window to shrink. Additionally, switching between tabs results in the window jumping around on the screen.

This behavior does not occur in other terminals, where tab management behaves as expected.

## Steps to Reproduce:

Open the terminal in a tiling window manager (e.g., AeroSpace or Amethyst).
Create a new tab.
Observe that the window shrinks.
Switch between tabs and observe that the window jumps to different positions.
Expected Behavior:
Tabs should be created and switched without causing window resizing or repositioning issues.

## Environment:

Terminal version: 
OS: 15.0.1
Window Manager: AeroSpace v0.16.0-Beta d172dfd8a92f2d339f3d46a12a297e43e80768ca

## Additional Context: 

This behavior is not observed in other terminals tested (e.g., iTerm2, stock Terminal, Alacritty).
Unsure if this is a terminal-specific issue or caused by the interaction with the tiling window manager.

**Comment by mitchellh** [comment, marked as answer]
This is well reported, and well documented how to fix it. Here is one resource: https://ghostty.org/docs/help/macos-tiling-wms
